examples/step-21/step-21.cc

```diff
@@ -512,8 +517,7 @@ namespace Step21
          1)
     , dof_handler(triangulation)
     , n_refinement_steps(5)
-    , time_step(0)
-    , timestep_number(1)
+    , time(/*start time*/ 0., /*end time*/ 1., /*time step*/ 0.)
```

Comment:
Shouldn't the `start_step_size` be non-zero?

Reply:
How about using the constructor I'm proposing in https://github.com/dealii/dealii/pull/9913?

Reply:
Setting the starting time step to zero is the idiomatic way (for this class) to say that we are going to provide the first time step size later in the code. I added more explanation and information about this in the documentation. I further modified the constructor such that the third parameter is optional and it is automatically set to zero if it is not provided by the user. Even though the class allows having `dt ==0`, it does not allow incrementing time when `dt == 0`, forcing the user to provide a positive (nonzero) step size before calling `advance_time()`.